### PR TITLE
fix: Correctly calculate diffs for CRDs with ignore overrides

### PR DESCRIPTION
**Pull request by leoluz**
Signed-off-by: Leonardo Luz Almeida <leonardo_almeida@intuit.com>

There was a problem when checking the ignore difference configuration for CRDs manifests. The problem is related to the fact that all CRDs will [be automatically configured](https://github.com/argoproj/argo-cd/blob/fa46ac178f809577285d8b37480ef1e46e7cb4a4/util/settings/settings.go#L733-L737) with resource overrides and that configuration was making application diff configs to be discarded. This PR will, instead, merge all diff configs available in settings with the Application's ones and use that merged object to determine how to calculate diffs.

This PR partially addresses the issue reported in https://github.com/argoproj/argo-cd/issues/9071

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: util/argo/diff/ignore_test.go [hunk 1]
```diff
@@ -187,5 +191,32 @@ func TestIgnoreDiffConfig_HasIgnoreDifference(t *testing.T) {
 		assert.False(t, ok)
 		require.Nil(t, actual)
 	})
+	t.Run("will merge ignore differences correctly", func(t *testing.T) {
```
Comment (crenshaw-dev) [inline util/argo/diff/ignore_test.go:194]: I don't think I follow this test... I think I see that we're supposed to de-dupe some ignore rules. Can you clarify?

Comment (leoluz) [reply util/argo/diff/ignore_test.go:194]: de-dupe during merge is what this test-case is validating. This is an additional logic improvement but main issue was that it was completely skipping the Application configs if any override config was present. The main scenario is validated in other test-cases.

**Review by crenshaw-dev — APPROVED** [review]
lgtm!

File: util/argo/diff/ignore_test.go (hunk 1 above)
Comment (leoluz) [reply util/argo/diff/ignore_test.go]: Improved the test-case description to make it more evident 👍🏻

**Pull request merged**